certification/internal/shell/engine.go

```diff
@@ -57,12 +63,16 @@ func (e *CheckEngine) ExecuteChecks() error {
 		if err != nil {
 			log.WithFields(log.Fields{"result": "ERROR", "error": err.Error()}).Info("check completed: ", check.Name())
 			e.results.Errors = append(e.results.Errors, runtime.Result{Check: check, ElapsedTime: checkElapsedTime})
+			e.results.Status = "ERROR"
```

Comment:
Thank you Emily!
IMO, the status should be one of passed or failed. 

In any case, I think if you manage setting the `status` field outside of the for loop, the code is easier to read/maintain; for example the first time I saw lines 73-75, it took me a while to understand why there is such a condition for setting the status to FAILED.

Reply:
Yeah I agree I will move setting the status outside the loop.

I guess I was wondering what exactly we want the status to indicate. It can either be an indicator of if it was successful or failed for whatever reason and the user can figure out what went wrong by looking at the check lists (Passed/Failed). Or it could represent the state of the policy execution (Passed/Failed/Error). Both are similar and I am fine either way.

Reply:
+1 to changes mentioned by @samira-barouti. They look implemented me so I'm resolving this thread. Just closing the loop with this message.